homeassistant/components/tado/device_tracker.py

```diff
@@ -1,149 +1,141 @@
-    return scanner if scanner.success_init else None
-
-
-Device = namedtuple("Device", ["mac", "name"])
 
+from .const import DATA, DOMAIN, SIGNAL_TADO_MOBILE_DEVICE_UPDATE_RECEIVED
 
-class TadoDeviceScanner(DeviceScanner):
-    """Scanner for geofenced devices from Tado."""
+_LOGGER = logging.getLogger(__name__)
 
-    def __init__(self, hass, config):
-        """Initialize the scanner."""
-        self.hass = hass
-        self.last_results = []
 
-        self.username = config[CONF_USERNAME]
-        self.password = config[CONF_PASSWORD]
+async def async_get_scanner(hass: HomeAssistant, config: ConfigType) -> None:
+    """Configure the Tado device scanner."""
+    async_create_issue(
+        hass,
+        DOMAIN,
+        "deprecated_yaml_import_device_tracker",
+        breaks_in_ha_version="2024.5.0",
+        is_fixable=False,
+        severity=IssueSeverity.WARNING,
+        translation_key="deprecated_yaml_import_device_tracker",
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Tado device scannery entity."""
+    _LOGGER.debug("Setting up Tado device scanner entity")
+    tado = hass.data[DOMAIN][entry.entry_id][DATA]
+    tracked: set = set()
+
+    async def async_update_devices() -> None:
+        """Update the values of the devices."""
+        add_tracked_entities(hass, tado, async_add_entities, tracked)
+
+    await async_update_devices()
+
+    entry.async_on_unload(
+        async_dispatcher_connect(
+            hass,
+            SIGNAL_TADO_MOBILE_DEVICE_UPDATE_RECEIVED,
+            async_update_devices,
+        )
+    )
+
+
+@callback
+def add_tracked_entities(
+    hass: HomeAssistant,
+    tado: Any,
+    async_add_entities: AddEntitiesCallback,
+    tracked: set[str],
+) -> None:
+    """Add new tracker entities from Tado."""
+    _LOGGER.debug("Fetching Tado devices from API")
+    new_tracked = []
+    for device_key, device in tado.data["mobile_device"].items():
+        if device_key in tracked:
+            continue
 
-        # The Tado device tracker can work with or without a home_id
-        self.home_id = config[CONF_HOME_ID] if CONF_HOME_ID in config else None
+        _LOGGER.debug(
+            "Adding Tado device %s with deviceID %s", device["name"], device_key
+        )
+        new_tracked.append(TadoDeviceTrackerEntity(device_key, device["name"], tado))
+        tracked.add(device_key)
+
+    async_add_entities(new_tracked)
+
+
+class TadoDeviceTrackerEntity(ScannerEntity):
+    """A Tado Device Tracker entity."""
+
+    def __init__(
+        self,
+        device_id: str,
+        device_name: str,
+        tado: Any,
+    ) -> None:
+        """Initialize a Tado Device Tracker entity."""
+        super().__init__()
+        self._device_id = device_id
+        self._device_name = device_name
+        self._tado = tado
+        self._active = False
```

Comment:
We should set `_attr_should_poll` class attribute above the init method to `False`, as this entity updates state via the dispatcher.